statsd-client-async/statsd_client_async.mli

```diff
@@ -0,0 +1,6 @@
+val gauge : ?sample_rate:float -> string -> int -> unit Async.Deferred.t
```

Comment:
I wonder if there's way to avoid having to copy paste this interface between async + lwt. This seems to be the standard way of handling these separations though (looking at `ocaml-cohttp` for inspiration) so maybe not.

Reply:
yeah, there is. Check out my dogstatsd pull request. I modify the async and lwt signatures into one liners for these functions.